pkg/apis/pipeline/v1alpha1/git_resource_test.go

```diff
@@ -100,6 +121,25 @@ func Test_Valid_NewGitResource(t *testing.T) {
 			Revision:   "test",
 			GitImage:   "override-with-git:latest",
 			Submodules: true,
+			Depth:      8,
+		},
+	}, {
+		desc: "With zero depth",
+		pipelineResource: tb.PipelineResource("git-resource", "default",
+			tb.PipelineResourceSpec(v1alpha1.PipelineResourceTypeGit,
+				tb.PipelineResourceSpecParam("URL", "git@github.com:test/test.git"),
+				tb.PipelineResourceSpecParam("Revision", "test"),
+				tb.PipelineResourceSpecParam("Depth", "0"),
+			),
+		),
+		want: &v1alpha1.GitResource{
+			Name:       "git-resource",
+			Type:       v1alpha1.PipelineResourceTypeGit,
+			URL:        "git@github.com:test/test.git",
+			Revision:   "test",
+			GitImage:   "override-with-git:latest",
+			Submodules: true,
+			Depth:      0,
```

Comment:
What actually happens when you clone with `--depth=0`? Is this a case we should catch and fail in validation? Should we reject negative values?

(It's probably fine to just pass through whatever and let `git` fail as it wants, but it can be useful to avoid running a container that we know will fail -- this validation also doesn't have to happen in this change)